ui/src/components/yaml/index.tsx

```diff
@@ -377,35 +380,45 @@
                   />
                 </Space>
               </div>
-              <div className={styles.yaml_content_diagnosisContent}>
-                {interpretStatus === 'loading' ||
-                (interpretStatus === 'streaming' && !interpret) ? (
-                  <div className={styles.yaml_content_diagnosisLoading}>
-                    <Spin />
-                    <p>{t('EventAggregator.DiagnosisInProgress')}</p>
-                  </div>
-                ) : interpretStatus === 'streaming' ? (
-                  <div>
-                    <Markdown>{interpret}</Markdown>
-                    <div className={styles.yaml_content_streamingIndicator}>
-                      <span className={styles.dot}></span>
-                      <span className={styles.dot}></span>
-                      <span className={styles.dot}></span>
+              <div className={styles.yaml_content_diagnosisBody}>
+                <div
+                  className={styles.yaml_content_diagnosisContent}
+                  ref={contentRef}
+                >
+                  {interpretStatus === 'loading' ||
+                  (interpretStatus === 'streaming' && !interpret) ? (
+                    <div className={styles.yaml_content_diagnosisLoading}>
+                      <Spin />
+                      <p>{t('EventAggregator.DiagnosisInProgress')}</p>
                     </div>
-                    <div
-                      ref={interpretEndRef}
-                      style={{ float: 'left', clear: 'both' }}
+                  ) : interpretStatus === 'streaming' ? (
+                    <>
+                      <Markdown>{interpret}</Markdown>
+                      <div
+                        ref={interpretEndRef}
+                        style={{
+                          float: 'left',
+                          clear: 'both',
+                        }}
+                      />
```

Comment:
Potential XSS vulnerability: Using {interpret} directly in JSX without sanitization. Consider using a sanitizer like marked or react-markdown-sanitizer.